app/routes/gather-data.js

```diff
@@ -51,87 +43,112 @@ module.exports = {
 }
 
 const processVetVisitsData = async (containerClient) => {
-  const manifestFilename = config.vetVisits.manifestFile
-  const manifestGrants = await getManifest(manifestFilename)
+  try {
+    const manifestFilename = config.vetVisits.manifestFile
+    const manifestGrants = await getManifest(manifestFilename)
+
+    const manifestData = [...manifestGrants]
 
-  const manifestData = [...manifestGrants]
+    const grants = await getVetVisits()
 
-  const grants = await getVetVisits()
+    const { chunkCount, processedGrants } = await process(grants, manifestData, 'Vet Visits', containerClient)
 
-  const { chunkCount, processedGrants } = await processGrants(grants, manifestData, 'Vet Visits', containerClient)
+    manifestData.push(...processedGrants)
 
-  manifestData.push(...processedGrants)
+    await uploadManifest(manifestData, manifestFilename, containerClient)
 
-  await uploadManifest(manifestData, manifestFilename, containerClient)
+    return {
+      chunkCount,
+      processedGrantsCount: processedGrants.length
+    }
+  } catch (error) {
+    console.error(error)
 
-  return {
-    chunkCount,
-    processedGrantsCount: processedGrants.length
+    return {
+      processedGrantsCount: 0
+    }
   }
 }
 
 const processWoodlandData = async (containerClient) => {
```

Comment:
ia it worth replacing with 1 function to simplify the logic and avoid mismatches oin logic?
```
const processGrantData = async ({ containerClient, manifestFilename, grantSchemeName, getGrants }) => {
  try {
    const manifestGrants = await getManifest(manifestFilename);
    const manifestData = [...manifestGrants];
    const grants = await getGrants();
    const { chunkCount, processedGrants } = await process(grants, manifestData, grantSchemeName, containerClient);

    manifestData.push(...processedGrants);
    await uploadManifest(manifestData, manifestFilename, containerClient);

    return {
      chunkCount,
      processedGrantsCount: processedGrants.length
    };
  } catch (error) {
    console.error(`Error processing ${grantSchemeName} data:`, error);
    return {
      processedGrantsCount: 0
    };
  }
};
```

Reply:
I've moved some of the logic into the processor (like uploading the manifest) so its shared between the schemes. Have left these functions as separate as there are differences between the schemes